### DISCUSSION TITLE
Can't get multiple records from a supabase function

### DISCUSSION THREAD

**Discussion by RobertBrunhage** [supabase/supabase#16915] [Contribute to Supabase]
### Discussed in https://github.com/supabase/supabase/discussions/4245

<div type='discussions-op-text'>

<sup>Originally posted by **RobertBrunhage** December  2, 2021</sup>
I have the following function that returns a `record` but this only seems to return the **first** record, instead of all of them.

![image](https://user-images.githubusercontent.com/26344867/144398717-9649af14-f50e-457d-ba01-217ad4b913ce.png)

I've tried the same query in the SQL editor/tester and during that it returns multiple records.

How do I make the function return all the records that satisfy the query?</div>

**Comment by phamhieu** [comment]
@RobertBrunhage the current function UI doesn't support returning a record set.

You need to use SQL editor to create the function cos you have to define the returning record definition. https://www.postgresql.org/docs/9.1/xfunc-sql.html#XFUNC-SQL-FUNCTIONS-RETURNING-TABLE

**Comment by EskelCz** [comment]
Also the UI in the /database/functions list doesn't indicate SETOF for the return value when already set

**Comment by Spiderpig86** [comment]
Bumping this issue since I think this would be a great addition for a lot of devs :)

**Comment by enricdema** [comment]
Hiii :)
I just did a function for my project and it works fine ;)

`CREATE FUNCTION ranking()
RETURNS TABLE(user_email varchar,score int) AS $$
    select scores.user_email, SUM(scores.score)
    from scores
    where "solved"='true'
    group by user_email
$$ LANGUAGE SQL;`
